src/main.rs

```diff
@@ -1,18 +1,6 @@
 use std::process::ExitCode;
 
-use clap::Parser;
-
 fn main() -> ExitCode {
-    // If none of the ldns-* tools matched, then we continue with clap
-    // argument parsing.
-    let env_args = std::env::args_os();
-    let args = dnst::try_ldns_compatibility(env_args).unwrap_or_else(dnst::Args::parse);
-
-    match args.execute() {
-        Ok(()) => ExitCode::SUCCESS,
-        Err(err) => {
-            err.pretty_print();
-            ExitCode::FAILURE
-        }
-    }
+    let env = dnst::env::real::RealEnv;
+    ExitCode::from(dnst::run(env))
```

Comment:
Why not this?

```suggestion
    dnst::run(env).into()
```

Reply:
In fact, why not use `ExitCode` instead of `u8`?

Reply:
Question 1: sure!

Question 2: `ExitCode` implements _nothing_, not `PartialEq`, not `TryInto<u8>`, just nothing except `Termination`. So we can't use it for a function that we want to check the output of.